### PR TITLE
ADD THE DATA MODELS AND LINTERS

### PR DESCRIPTION
**Pull request by Ochiengsteven**
### Changes Made

- [x] New app created and named appropriately.
- [x] Git initialized for version control. 🌳
- [x] Postgres database set up. 🔄🐘
- [x] Project schema built according to ERD diagram.
- [x] Migration files created and executed successfully. ⚙️
- [x] Table and column names aligned with specifications.
- [x] 'photo' column in 'users' table linked to an image. 📸
- [x] Foreign key column in 'posts' table named 'author_id'. 🤝
- [x] Foreign keys included with corresponding indexes. 📊
- [x] References from sneak peek followed for clarity.

### Additional Notes

- Followed the guidelines and references from the sneak peek to ensure accurate table and column names.

### Testing

1. Manually verified that the database schema matches the expected structure.

Please review and allow merge.

Thank you! 🙌

### CODE REVIEW

File: db/migrate/20231129094010_create_comments.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateComments < ActiveRecord::Migration[7.1]
+  def change
+    create_table :comments do |t|
+      t.integer :author_id
+      t.integer :post_id
```
Comment (MahmoudBakr23) [inline db/migrate/20231129094010_create_comments.rb:5]: - [ ] According to the [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png), please keep in mind that the author_id column is only used with the posts table, and not in the comments and likes tables, please stick to the given ERD for better a development experience. 🙏

File: db/migrate/20231129094036_create_likes.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateLikes < ActiveRecord::Migration[7.1]
+  def change
+    create_table :likes do |t|
+      t.integer :author_id
+      t.integer :post_id
```
Comment (MahmoudBakr23) [inline db/migrate/20231129094036_create_likes.rb:5]: - [ ] According to the [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png), please keep in mind that the author_id column is only used with the posts table, and not in the comments and likes tables, please stick to the given ERD for better a development experience. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @Ochiengsteven 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on making the migrations' files correctly so far
- Nice work with the indexes
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Ochiengsteven, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**